libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -612,13 +612,38 @@ bool DeclarationRegistrationHelper::visitNode(ASTNode& _node)
 
 	if (auto* declaration = dynamic_cast<Declaration*>(&_node))
 		registerDeclaration(*declaration);
+
+	// Setting the canonicalname before ScopeOpener, so that m_currentScope is not same as the
```

Comment:
Not sure this comment is actually needed.

Reply:
Can remove it.